src/DIRAC/ConfigurationSystem/Client/VOMS2CSSynchronizer.py

```diff
@@ -353,6 +353,8 @@ def syncCSWithVOMS(self):
       keepGroups = nonVOGroups + groupsWithRole
       if not quarantineVOGroup or quarantineVOGroup not in existingGroups:
         keepGroups += [defaultVOGroup]
+      if quarantineVOGroup and quarantineVOGroup in existingGroups:
+        keepGroups = [quarantineVOGroup]
```

Comment:
The logic of this Synchronier is too convoluted for me to follow quickly, but is setting keepGroups to only the quarantine group here intentional because in the next loop additional groups might be added to keepGroups again. So if _only_ the quarantineGroup should be keepGroups this should be done after the following for loop?

Reply:
The logic is indeed way too convoluted. It would deserve a big refactoring but not right now.

In the subsequent loop, the synchronizer will add to `keepGroups` those groups that do not come from VOMS (basically, those that were previously assigned "by hand"). Basically being in `quarantineGroup` will exclude you from being synced with VOMS.

Reply:
But is that really what we want ? @andresailer logic seems more appropriate to me. If you are in quarantine, you are in quarantine only, full stop.
You could just indent the whole `for loop` below and add an `else` statement ?

Reply:
Well, we might have a case when user *pippo* is in the groups: 
- "vo_users" which has an VOMSRole = /vo/Role=user
- "some_internal_to_DIRAC_group" with no VOMS role

We define a DIRAC group "vo_low_prio_users", and we want users in there to still be able to submit jobs, so we also give to this group the VOMSRole = /vo/Role=user. 

If we put user *pippo* to group "vo_low_prio_users", from which group do we want him to be removed from?
1. all but "vo_low_prio_users"
2. only from "vo_users"
?

The advantage of 2 is that of course we don't have to remember that, once *pippo* is removed from "vo_low_prio_users", we should also re-add him to "some_internal_to_DIRAC_group". This is the `VOMS2CSSynchronizer` so I don't think it should touch groups that are not VOMS related.

Reply:
yes that makes sense... 
Though I am thinking that it would probably make the logic simpler if we added an boolean attribute `exclusive` to a group. A user in that group is removed from all other groups (so more like your option 1.) , and an error is thrown if a user is in more than one `exclusive` group. This makes the logic of quarantine/banned/etc more generic, and probably easier to manage

Reply:
Just update the PR title and release notes to "remove it from all other VOMS groups" so that the code matches the change description.

Reply:
done

Reply:
review OK now?